kusto/TokenProvider.go

```diff
@@ -0,0 +1,25 @@
+package kusto
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+)
+
+type tokenProvider struct {
+	tokenCredential azcore.TokenCredential

```

Comment:
add an explanation of the attributes. Is the file to be named tokenprovider.go ?